ios-xcframework/build.sh

```diff
@@ -18,13 +18,130 @@ function _xcodebuild {
+
+rm -rf "$ARCHIVES_ROOT"
+rm -rf "$FINAL_OUTPUT"
+
+# 1. Archive for iOS
+log "Archiving $SCHEME for iOS"
+
+_xcodebuild archive \
+  -workspace "$WORKSPACE" \
+  -scheme "$SCHEME" \
+  -configuration Release \
+  -sdk iphoneos \
+  -archivePath "$IOS_DEVICE_ARCHIVE_PATH" \
+  -derivedDataPath "$DERIVED_DATA_PATH" \
+  BUILD_LIBRARY_FOR_DISTRIBUTION=YES \
+  SKIP_INSTALL=NO
+
+# 2. Archive for Simulator
+log "Archiving $SCHEME for Simulator"
+
+_xcodebuild archive \
+  -workspace "$WORKSPACE" \
+  -scheme "$SCHEME" \
+  -configuration Debug \
+  -sdk iphonesimulator \
+  -archivePath "$IOS_SIMULATOR_ARCHIVE_PATH" \
+  -derivedDataPath "$DERIVED_DATA_PATH" \
+  BUILD_LIBRARY_FOR_DISTRIBUTION=YES \
+  SKIP_INSTALL=NO
+
+# 3. Create XCFrameworks for every framework in the archives
+XCFRAMEWORKS_DIR=xcframeworks
+
+rm -rf $XCFRAMEWORKS_DIR
+mkdir -p $XCFRAMEWORKS_DIR
+
+for FRAMEWORK in $(find $IOS_DEVICE_ARCHIVE_PATH/Products/Library/Frameworks -type d -name "*.framework");
+do
+  CURRENT_FRAMEWORK_NAME=$(basename "$FRAMEWORK" .framework)
+
+  # TODO: Use a framework list instead of building all of them?
+
+  # 1. Create XCFrameworks for every framework that was generated by the main
+  # framework build process
+  FRAMEWORK_RELATIVE_PATH="Products/Library/Frameworks/$CURRENT_FRAMEWORK_NAME.framework"
+
+  log 'package' "Creating XCFramework for $CURRENT_FRAMEWORK_NAME"
+
+  _xcodebuild \
+    -create-xcframework \
+    -framework "$IOS_DEVICE_ARCHIVE_PATH/$FRAMEWORK_RELATIVE_PATH" \
+    -framework "$IOS_SIMULATOR_ARCHIVE_PATH/$FRAMEWORK_RELATIVE_PATH" \
+    -output "$XCFRAMEWORKS_DIR/$CURRENT_FRAMEWORK_NAME.xcframework"
+done
+
+log 'compression' 'Zipping Gutenberg XCFrameworks'
+ZIP_PATH=xcframeworks/Gutenberg.zip
+zip -rq "$ZIP_PATH" \
+  xcframeworks/Aztec.xcframework \
+  xcframeworks/Gutenberg.xcframework \
+  xcframeworks/React.xcframework \
+  xcframeworks/RNTAztecView.xcframework \
+  xcframeworks/yoga.xcframework
+echo "Gutenberg XCFrameworks ZIP generated at $ZIP_PATH"
+
+# In parallel to the project to ship Gutenberg as an XCFramework we are also
+# experimenting with adding React Native into other apps with an XCFramework.
+#
+# For convenience, we produce a ZIP with all the React Native XCFrameworks
+# here, even though it would be more efficient to have a dedicated project that
+# mirrors React Native and builds new one whenever a new version is released.
+#
+# In this current implementation, the approach is wasteful because we are
+# publishing the same ZIP every time. A possible imporvement would be to track
+# the RN version and check against the storage medium to see if an archive for
+# that version is already available.
+log 'compression' 'Zipping React XCFrameworks'
+ZIP_PATH=xcframeworks/ReactNative.zip
+zip -rq "$ZIP_PATH" xcframeworks -x \
+  xcframeworks/Aztec.xcframework \
+  xcframeworks/Gutenberg.xcframework \
+  xcframeworks/RNTAztecView.xcframework \
+  xcframeworks/Pods_Gutenberg.xcframework
```

Comment:
Is it correct that we package up the `Pods_Gutenberg` framework? In my understading, this "Pods" framework should has almost no code and just declares some dependencies?

Reply:
Your understanding is correct.

But in this instance, it's appropriate to exclude `Pods_Gutenberg` because this `zip` command is for the React Native XCFrameworks, and the list is that of the `.xcframework` to exclude (`-x` parameter) from the archive.

Reply:
By the way, I think we'll get rid of this archiving step soon since it seems like https://github.com/woocommerce/WooCommerce-Shared can do what we need without the need to import these.

Reply:
Sorry, I missed the `-x` option 🥲